includes/Story_Post_Type.php

```diff
@@ -471,28 +448,108 @@ public static function admin_body_class( $class ) {
 	 * @return array Allowed tags.
 	 */
 	public static function filter_kses_allowed_html( $allowed_tags ) {
-		if ( ! class_exists( '\AMP_Allowed_Tags_Generated' ) ) {
-			return $allowed_tags;
-		}
-
 		$story_components = [
-			'amp-story',
-			'amp-story-page',
-			'amp-story-grid-layer',
-			'amp-story-cta-layer',
-			'amp-story-page-attachment',
-			'amp-img',
-			'amp-video',
-			'img',
+			'amp-story'                 => [
+				'background-audio'     => true,
+				'live-story'           => true,
+				'live-story-disabled'  => true,
+				'poster-landscape-src' => true,
+				'poster-portrait-src'  => true,
+				'poster-square-src'    => true,
+				'publisher'            => true,
+				'publisher-logo-src'   => true,
+				'standalone'           => true,
+				'supports-landscape'   => true,
+				'title'                => true,
+			],
+			'amp-story-page'            => [
+				'auto-advance-after' => true,
+				'background-audio'   => true,
+				'id'                 => true,
+			],
+			'amp-story-page-attachment' => [
+				'theme' => true,
+			],
+			'amp-story-grid-layer'      => [
+				'position' => true,
+				'template' => true,
+			],
+			'amp-story-cta-layer'       => [],
+			'amp-img'                   => [
```

Comment:
Is the list copied over from the automatically generated tags? Wanted to ask if `layout` is missing from here, however, testing shows that saving works as-is right now for layout.

Should these attributes be pulled automatically eventually again?

Reply:
> Is the list copied over from the automatically generated tags?

Yes.

> Wanted to ask if `layout` is missing from here

No. `layout` is added a few lines further below to all the elements. See: https://github.com/google/web-stories-wp/pull/140/files#diff-99d793dc332da60eaf6b6846bfd3a7f0R554-R560

> Should these attributes be pulled automatically eventually again?

Ideally yes, but parsing the whole AMP spec for this is way overkill for now.  Plus, I don't expect this to change too much. Perhaps we can eventually have a super-stripped-down version of https://cdn.ampproject.org/v0/validator.json (see https://github.com/ampproject/amp-wp/issues/2769)

Reply:
Agreed, we definitely shouldn't have the whole AMP spec for this, however, we can adjust the python script to only save information about the relevant components to the PHP file. I have quite some experience with python (a long time ago though) and IIRC @barklund as well (likely more than me).

Reply:
Let's keep an eye on what the AMP plugin is doing with the JSON version too, and also keep an eye on the progress on extracting everything into a separate library. That would make it easier for everyone else on the team to maintain as well.